rp-binary-info/src/types.rs

```diff
@@ -189,4 +189,36 @@ impl IntegerEntry {
     }
 }
 
+/// An alias for IntegerEntry, taking a pointerinstead of an integer
+#[repr(C)]
+pub struct PointerEntry {
+    header: EntryCommon,
+    id: u32,
+    value: *const (),
+}
+
+impl PointerEntry {
+    /// Create a new `StringEntry`
+    pub const fn new(tag: u16, id: u32, value: *const ()) -> PointerEntry {
+        PointerEntry {
+            header: EntryCommon {
+                data_type: DataType::IdAndInt,
```

Comment:
We should point out we're relying on an int and a pointer to have the same representation - there's an implicit transmute happening here to bypass an issue with pointers in const functions.